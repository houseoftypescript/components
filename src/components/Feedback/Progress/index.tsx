import React from 'react';
import styled from 'styled-components';

const ProgressContainer = styled.div`
  border-radius: 0.25rem;
  height: 1rem;
  overflow: hidden;
  background-color: #d1d5db;
`;

type ProgressBarProps = { value: number };

const ProgressBar = styled.div<ProgressBarProps>`
  height: 1rem;
  overflow: hidden;
  background-color: #ef4444;
  transition: width linear 100ms;
  width: ${(props: ProgressBarProps) => props.value}%;
`;

const Progress: React.FC<{ value: number }> = ({ value = 0 }) => {
  return (
    <ProgressContainer>
      <ProgressBar value={value} />
    </ProgressContainer>
  );
};

export default Progress;
