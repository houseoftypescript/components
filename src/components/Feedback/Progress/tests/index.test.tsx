import { render } from '@testing-library/react';
import React from 'react';
import Progress from '../index';

describe('Progress', () => {
  it('to match snapshot', () => {
    const { container } = render(<Progress value={10} />);
    expect(container).toMatchSnapshot();
  });
});
