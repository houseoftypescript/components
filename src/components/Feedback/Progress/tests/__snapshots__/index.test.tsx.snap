// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress to match snapshot 1`] = `
<div>
  <div
    class="sc-beySbM iilZOj"
  >
    <div
      class="sc-guDLRT jzTdXj"
      value="10"
    />
  </div>
</div>
`;
